barretenberg/cpp/src/barretenberg/crypto/pedersen_hash/pedersen_refactor.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+// TODO(@zac-wiliamson #2341 rename to pedersen.hpp once we migrate to new hash standard)
+
+#include "../generators/fixed_base_scalar_mul.hpp"
+#include "../generators/generator_data.hpp"
+#include "../pedersen_commitment/pedersen_refactor.hpp"
+#include "barretenberg/ecc/curves/grumpkin/grumpkin.hpp"
+#include <array>
+
+namespace crypto {
+
+/**
+ * @brief Performs pedersen hashes!
+ *
+ * To hash to a size-n list of field elements `x`, we return the X-coordinate of:
+ *
+ *      Hash(x) = n.[h] + Commit(x)
+ *
+ * Where `g` is a list of generator points defined by `generator_data`
+ * And `h` is a unique generator whose domain separator is the string `pedersen_hash_length`.
+ *
+ * The addition of `n.[h]` into the hash is to prevent length-extension attacks.
+ * It also ensures that the hash output is never the point at infinity.
```

Comment:
Note to self: need to come back to this to double check that the output of Commit(0) is well defined

Reply:
Commit(0) *should* return the point at infinity. I think we have a test for this effect in `cycle_group.test.cpp`

Hash(0) will produce a valid output because of the `n.[h]` term added into the hash